### ISSUE TITLE
CLI driver (SSH) on Nexus might not discover all arp entries

### ISSUE THREAD

**Issue by j-hoff** [netdisco/netdisco#1221]
Hi, we ran into an issue when we configured collection via SSH on our Nexus (N77c7706) devices.

## Expected Behavior
When doing an arpnip on the devices all arp entries should be discovered.

## Current Behavior
In our environment only a few arp entries (21 of 3800) are discovered. 
There is no hint of what goes wrong in the regular debug output.

```
[209755] 2024-06-11 13:03:22 debug ⮕ worker Arpnip::Nodes p200
[209755] 2024-06-11 13:03:22 debug cli session cache warm: [xxx.xxx.xxx.xx]
[209755] 2024-06-11 13:03:26 debug xxx.xxx.xxx.xx 209755 arpnip()
[209755] 2024-06-11 13:03:46 debug ⬅ (done) Gathered arp caches from xxx.xxx.xxx.xx
```
After turning on debug of the Expect module in `SSHCollector/Platform/NXOS.pm`
the error appears to be with switching off the terminal paging:
```
`terminal length 0\r\r\n% Permission denied for the role\r\n\r
```
then only one page (of 21 entries) is read from the arp table, output ends with
```
--More--
```

## Possible Solution
We wanted to give the user as few rights as possible, so we decided to add  `| no-more` 
to the two expect calls in `SSHCollector/Platform/NXOS.pm`.
In my understanding setting `terminal length 0` would be unnecessary then and this solution works even if the user has no rights to change the terminal settings.

`$expect->send("show ip arp vrf all | inc : | no-more\n");`

`$expect->send("show ipv6 neighbor vrf all | exclude Flags: | no-more\n");`

I would also be happy to provide a PR.

## Your Environment

* Netdisco version used:  	2.76.5
* SNMP::Info version used: 3.970.1

## Config info (deployment.yml)
```
- tag: nexusssh
    driver: cli
    platform: NXOS
    only: 
       - 'model:N77c7706'
    username: xxx
    password: yyy
```

**Comment by rc9000** [comment]
Hi! 

I tried this no-more variant on a few quite old and quite new NX-OS releases and it seems to work fine on all of them.

From a philosophical standpoint it seems a bit weird that your org went through the trouble of configuring RBAC, giving an user all the show commands for this module, but then draws the line at not allowing "terminal length 0". So I would recommend to fix it there instead. However this might be outside of your power, not your decision, a big company yadda yadda yadda... If you put it in a PR we'll probably merge it unless somebody else is strongly against.

In any case, thanks already for figuring it out and posting a solution.

**Comment by j-hoff** [comment]
Thanks! I totally get your point and agree mostly.

I was not directly involved in setting up the access user and my knowledge in the matter is limited,
but I had a chat with the colleagues now and it appears that  the user has _priviledge level 1_ set via radius, which is predefined for users with "very limited read-only access". Only levels 0,1 and 15 seem to be defined by default, so level 1 was the obvious choice for them.
It was not a conscious decision to disallow the `terminal length 0` command.

We could probably change it to allow the `terminal length` command.
But if you don't mind going with the `| no-more` solution I think it would be a good idea as others might run into the same problem.

**Comment by ollyg** [comment]
implemented in dbe2d7e0

**Comment by ollyg** [comment]
fixed in dbe2d7e0
